### ISSUE TITLE
Change cursor style over Map

### ISSUE THREAD

**Issue by fitoprincipe** [jupyter-widgets/ipyleaflet#167]
It would be great if somehow the map's cursor could be changed to any of the supported styles. It is not hard to do it on the JavaScript side, just changing Map's `cursor` style ([stackoverflow](https://stackoverflow.com/questions/37518278/leaflet-how-can-i-change-the-grabbing-hand-cursor/41232864?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)), but I wouldn't know how to do it on ipyleaflet widget

**Comment by SylvainCorlay** [comment]
OK, I think that we need to have a `style` widget on the map just like on the ipywidgets slider to set CSS properties of the map.

**Comment by fitoprincipe** [comment]
That would be great! thank you

**Comment by fitoprincipe** [comment]
Hi @martinRenou. Any chance to include this in the Project's TODO?

**Comment by martinRenou** [comment]
Sure. But it doesn't mean I'll fix it soon... I added "simple" issues to the TODO list, issues that are newcomers friendly.

**Comment by fitoprincipe** [comment]
I understand, thank you @martinRenou, I wish I'd have some time to learn how all works (specially the JS side) to fix it myself. I've been reading but I'm not there yet. =)

**Comment by fitoprincipe** [comment]
I need some guide (help) here..

In  `leaflet.py` I made 2 changes:

1. A new class:

``` python
class MapStyle(Style, CoreWidget, Widget):
    """ Map Style Widget """
    _model_name = Unicode('MapStyleModel').tag(sync=True)
    cursor = Unicode(help='Cursor type').tag(sync=True)
```
2. imported `InstanceDict` from `ipywidgets.widgets.trait_types` and add an attribute in `Map`: 

``` python
style = InstanceDict(MapStyle).tag(sync=True, **widget_serialization)
```

In `Map.js` I created a new class:

``` javascript
var MapStyleModel = widgets.WidgetModel.extend({
    defaults: _.extend({}, widgets.StyleModel.prototype.defaults, {
        _model_name : "MapStyleModel",
        _model_module : "jupyter-leaflet",
        }),
 
    styleProperties = {
        cursor: {
            selector: '.leaflet-container',
            attribute: 'cursor',
            default: 'move'
        }
    }
}); 
```
I added this new class in `exports`:

``` javascript
module.exports = {
  LeafletMapView : LeafletMapView,
  LeafletMapModel : LeafletMapModel,
  MapStyleModel: MapStyleModel
};
```
 and also modified `jupyter-leaflet.js`:

``` javascript
//Exports
module.exports = {
...
MapStyleModel : map.MapStyleModel,
};
```

But I encounter an error:

![error_stylemap](https://user-images.githubusercontent.com/10751505/56661951-3e12b480-6679-11e9-88de-a15e1274fb8a.png)

**Comment by martinRenou** [comment]
Awesome! 
I'm not sure, but I guess the issue is that you are missing a `_model_module : Unicode("jupyter-leaflet")` in the Python model.

**Comment by martinRenou** [comment]
I'm also not sure the inheritance from `CoreWidget` is the right thing to do, given the name it does not feel right because it is not an `ipywidgets` core widget. I don't know what would be the right thing to do though, never implemented a custom Style before.

**Comment by fitoprincipe** [comment]
Thank you @martinRenou!, I'll check all those. I'm copying a lot from `ipywidgets`, so I may make mistakes due to it..

**Comment by fitoprincipe** [comment]
New error:

![error_stylemap2](https://user-images.githubusercontent.com/10751505/56662892-57b4fb80-667b-11e9-8550-312e075ef0b0.png)

**Comment by martinRenou** [comment]
Oh, I think I know what's happening here. You are copying code from ipywidgets which is written in TypeScript, while ipyleaflet is written in pure JavaScript. You would need to add the `styleProperties` static member in a different way, I guess by replacing the 
```
styleProperties = {
    ...
}
```
by
```
styleProperties: {
    ...
}
```

**Comment by fitoprincipe** [comment]
You are right! I was aware that `ipywidget` js side was in TypeScript (another complication for me), but missed that detail.. now I have a new error :laughing: 

![error_stylemap3](https://user-images.githubusercontent.com/10751505/56663658-fee66280-667c-11e9-9e1b-2d4a18553982.png)

at least the map is showing now =)

**Comment by fitoprincipe** [comment]
Ok, now I changed this:

``` javascript
var MapStyleModel = widgets.WidgetModel.extend({
   ...
});
```
for this:

``` javascript
var MapStyleModel = widgets.StyleModel.extend({
  ...
});
```

and no errors are shown, but it doesn't work.. any chance @SylvainCorlay or @jasongrout can have a quick look? This may be a minor thing, since I have almost zero knowledge about how the system works..

**Comment by martinRenou** [comment]
Do you mind opening a PR with what you have currently? So that we have a look at the entire thing?

**Comment by fitoprincipe** [comment]
I don't mind at all! since I am a beginner on this, I was afraid of opening a dumb PR.. but I guess if it is so dumb it'll be rejected at the end.. which I don't mind either! this is a learning process to me.. =)

**Comment by martinRenou** [comment]
Don't be! It's really cool that you are trying to contribute :)

**Comment by fitoprincipe** [comment]
Thanks to being open to newbies =) I have a small problem with the PR.. I first commited a change to `.gitignore` since I use IntelliJ IDEA IDE and it creates a junk of config files... but when I try to make the PR I cannot find the way to exclude that change..

**Comment by martinRenou** [comment]
You can try to remove your previous commit with `git reset HEAD~`, this will undo the commit. Then commit again without commiting the change on `.gitignore`

**Comment by fitoprincipe** [comment]
How do I close this making a reference to the PR?

**Comment by martinRenou** [comment]
You can put a comment like `fixes #167` or `closes #167` directly in the PR that will automatically close the issue when the PR is merged. But your comment in the PR was `Trying to solve issue #167`, github did not understand it.

Now you can just mention the PR in a comment with the # followed by the PR number, and then close it :) I let you do it

**Comment by fitoprincipe** [comment]
Great! Thanks again =)

**Comment by fitoprincipe** [comment]
Closed by #339
